Off-Top-App/getUserService
# OT-InstantiateKafkaServices | Create Producer & Consumer so that other microservices can query user data

WHAT I DID:
1. Created a Consumer so that a query can be made to the User Database and sent back to any microservices requesting User Data through a Producer.

HOW TO TEST:
1. You will need to create the topic manually on your machine before you run the services. To do so, open your command line and cd into the kafka directory on your machine and run the command:
 ```bin/kafka-topics.sh --zookeeper 127.0.0.1:2181 --topic userDataResponse --create --partitions 1 --replication-factor 1```
2. You will need zookeeper/kafka running. Then have this microservice running.
3. You will need to have two files from this repo (https://github.com/Off-Top-App/UserReportService) running. If you do not have this microservice setup yet then you will need to following:
```
- Depencencies: Have Node V12.10.0 installed.
- Pre-req for application: run `npm install`
- How to Run: node <filname> and the recommended way to run the service is like so: nodemon <filename>
```
4. Make sure you have `consumer.app.js` running first and then have `producer.app.js` run; make sure that UserService is running before you run `producer.app.js` because this is when the communication between these microservices begins.
5.  If everything went well you should see the following response in the terminal where you ran `node consumer.app.js`
<img width="1415" alt="Screen Shot 2020-06-24 at 2 33 32 AM" src="https://user-images.githubusercontent.com/26909882/85534934-9a979980-b5c6-11ea-9096-dea547ee856b.png">
